File: Patrones/Estructurales/Composite/Python/Supervisor.py
from Empleado import Empleado

class Supervisor(Empleado):
    def __init__(self, nombre, puesto):
        self.__nombre = nombre
        self.__puesto = puesto
        self.__subordinados = []

    def agregarSubordinado(self, empleado):
        self.__subordinados.append(empleado)

    def removerSubordinado(self, empleado):
        self.__subordinados.remove(empleado)

    def mostrarDetalles(self):
        print(f"Supervisor: {self.__nombre}, Puesto: {self.__puesto}")
        print("Subordinados:")
        
        for subordinado in self.__subordinados:
            subordinado.mostrarDetalles()  # Recursivamente muestra los detalles de subordinados

        print()

File: Patrones/Estructurales/Composite/Java/src/Supervisor.java
import java.util.ArrayList;
import java.util.List;

public class Supervisor implements Empleado {
    private final String nombre;
    private final String puesto;
    private final List<Empleado> subordinados;

    public Supervisor(String nombre, String puesto) {
        this.nombre = nombre;
        this.puesto = puesto;
        subordinados = new ArrayList<>();
    }

    public void agregarSubordinado(Empleado empleado) {
        subordinados.add(empleado);
    }

    public void removerSubordinado(Empleado empleado) {
        subordinados.remove(empleado);
    }

    @Override
    public void mostrarDetalles() {
        System.out.println("Supervisor: " + nombre + ", Puesto: " + puesto);
        System.out.println("Subordinados:");
        for (Empleado subordinado : subordinados) {
            subordinado.mostrarDetalles();  // Recursivamente muestra los detalles de subordinados
        }
    }
}
